### PR TITLE
OPSEXP-1980: Add support for importing certificates in the JRE

### PR DESCRIPTION
**Pull request by alxgomz**
Ref: OPSEXP-1980

### CODE REVIEW

File: roles/java/meta/argument_specs.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          path:
+            description: |
+              Local path to the java keystore
+            required: true
+            type: str
+          pass:
+            description: |
+              Password to unlock the java keystore
+            required: true
+            type: str
```
Comment (gionn) [inline roles/java/meta/argument_specs.yml:18]: not sure we could have this as a role argument, how it fits with the pre-existing ansible-vault feat?

File: roles/java/meta/argument_specs.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+              path:
+                type: str
+                description: |
+                  Local path to the host specific PKCS12 certificate container
+                required: true
+              pass:
+                type: str
+                description: |
+                  Certificate container password
+                required: true
```
Comment (gionn) [inline roles/java/meta/argument_specs.yml:40]: 🤔

File: roles/java/meta/argument_specs.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+              name:
+                type: str
+                description: |
+                  Name of the security key to  use as an alias in the keystore
+                required: true
+              pass:
+                type: str
+                description: |
+                  Passphrase of the security key
+                required: true
```
Comment (gionn) [inline roles/java/meta/argument_specs.yml:62]: 🤔

File: roles/java/meta/argument_specs.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        required: false
+        description: |
+          Local path to the java truststore
+        default: "{{ java_home }}/lib/security/cacerts"
+      java_truststore_pass:
+        type: str
+        required: false
+        description: |
+          Password to unlock the java truststore
+        default: changeit
```
Comment (gionn) [inline roles/java/meta/argument_specs.yml:88]: 🤔

File: roles/java/meta/argument_specs.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  main:
+    short_description: Main entrypoint for the java role
+    options:
+      java_keystore:
+        type: dict
+        options:
+          path:
+            description: |
+              Local path to the java keystore
+            required: true
```
Comment (gionn) [inline roles/java/meta/argument_specs.yml:12]: probably not, if user doesn't requires mtls

**Review by gionn — COMMENTED** [review]
I need more docs to better review, I am unsure how this will fit with the main playbook

File: roles/java/meta/argument_specs.yml (hunk 1 above)
Comment (alxgomz) [reply roles/java/meta/argument_specs.yml:18]: What I had in mind was:
* a we add a new secret in the `Populate secrets.yml with missing secrets` task (e.g.  `keystore_password`)
* pass that var as a role argument when calling the java role (`repository` molecule converge example below).

```yaml
- name: Converge
  hosts: all
  roles:
    - role: postgres
    - role: java
      java_keystore:
        type: JKS
        path: "{{ content_folder }}/keystore/alfresco.jks"
        pass: "{{ keystore_password }}"
        cert_containers:
          - path: "{{ inventory_hostname }}.p12"
            pass: "{{ keystore_password }}"
            trustcachain: true
    - role: activemq
    - role: repository
```

File: roles/java/meta/argument_specs.yml (hunk 5 above)
Comment (alxgomz) [reply roles/java/meta/argument_specs.yml:12]: This use case was declined in https://alfresco.atlassian.net/wiki/spaces/TECH/pages/1124696953/Certificate+deployment+in+Ansible+playbook

File: roles/java/meta/argument_specs.yml (hunk 2 above)
Comment (alxgomz) [reply roles/java/meta/argument_specs.yml:40]: p12 are always protected with passwords

File: roles/java/meta/argument_specs.yml (hunk 4 above)
Comment (alxgomz) [reply roles/java/meta/argument_specs.yml:88]: for compatibility purpose with the default JDK provided truststore

**Comment by alxgomz** [comment]
@gionn I tried addressing your comments and added a simple readme to the role. Let me know if that is sufficient.

File: roles/java/meta/argument_specs.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                required: true
+              pass:
+                type: str
+                description: |
+                  Certificate container password
+                required: true
+              trustcachain:
+                type: bool
+                description: |
+                  Whether to look for and trust CA chain
```
Comment (gionn) [inline roles/java/meta/argument_specs.yml:44]: what about something more explicit like `add_to_trusted_ca`?

**Review by gionn — APPROVED** [review]

**Pull request merged**